networkx/algorithms/cycles.py

```diff
@@ -1150,3 +1152,67 @@ def _path_to_cycle(path):
         # Toggle whether to keep the current edge.
         edges ^= {edge}
     return edges
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def girth(G):
+    """Returns the girth of the graph.
+
+    The girth of a graph is the length of its shortest cycle, or infinity if
+    the graph is acyclic. The algorithm follows the description given on the
+    Wikipedia page [1], and runs in time O(mn) on a graph with m edges and n
+    nodes.
+
+    Parameters
+    ----------
+    G : NetworkX Graph
+
+    Returns
+    -------
+    An int, or infinity.
+
+    Examples
+    --------
+    All examples below (except P_5) can easily be checked using Wikipedia,
+    which has a page for each of these famous graphs.
+
+    >>> girth(nx.chvatal_graph())
+    4
+    >>> girth(nx.tutte_graph())
+    4
+    >>> girth(nx.petersen_graph())
+    5
+    >>> girth(nx.heawood_graph())
+    6
+    >>> girth(nx.pappus_graph())
+    6
+    >>> girth(nx.path_graph(5))
+    inf
+
+    References:
+        [1] https://en.wikipedia.org/wiki/Girth_(graph_theory)
+
+    """
+    len_shortest_cycle = inf
+    all_edges = set(map(frozenset, G.edges))
```

Comment:
Just a nit, but comprehensions are usually more readable than `map`:

```suggestion
    all_edges = {frozenset(e) for e in G.edges}
```

Feel free to ignore.

Reply:
OK for me.